flink-connector-cassandra/src/main/java/org/apache/flink/connector/cassandra/source/reader/CassandraRow.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.cassandra.source.reader;
+
+import org.apache.flink.connector.cassandra.source.split.RingRange;
+
+import com.datastax.driver.core.ExecutionInfo;
+import com.datastax.driver.core.Row;
+
+/**
+ * Wrapper for Cassandra {@link Row} that stores associated {@link RingRange} to be able to update
+ * split states. It also stores {@link ExecutionInfo} Cassandra statistics about the query execution
+ * that produced this row.
+ */
+public class CassandraRow {
+
+    private final Row row;
+    private final RingRange associatedRingRange;
+    private final ExecutionInfo executionInfo;
```

Comment:
Who is actually using this? I see we pass it to the result set when creating the mapper; is it actually relevant there?

Reply:
Yes otherwise the mapper fails with an exception when we try to map

Reply:
Now that we store the `ResultSet` in `CassandraSplitState` to keep track of the fetch state, we no more need this `CassandraRow` wrapper as will have the `ExecutionInfo` in the `Resultset` with the call to `RecordEmitter#emitRecord()`